File: backend/stats/src/UrlShortener.Stats.Domain/Entities/UrlStat.cs
using UrlShortener.Stats.Domain.Entities.Interfaces;

namespace UrlShortener.Stats.Domain.Entities;

public class UrlStat(
	string id,
	string originalUrl,
	int accessesQuantity,
	DateTime lastAccess
) : IUrlStat
{
	public string Id { get; private set; } = id;

	public string OriginalUrl { get; private set; } = originalUrl;

	public int AccessesQuantity { get; private set; } = accessesQuantity;

	public DateTime LastAccess { get; private set; } = lastAccess;

	public void AddAccess()
	{
		AccessesQuantity += 1;
		LastAccess = DateTime.Now;
	}
}

File: backend/stats/internal/domain/url_stat.go
package domain

import "time"

type UrlStat struct {
	Id               string
	OriginalUrl      string
	AccessesQuantity int
	LastAccess       time.Time
}

func (u *UrlStat) AddAccess() {
	u.AccessesQuantity += 1
	u.LastAccess = time.Now()
}
